### PR TITLE
[Bugfix] Fix bug task process database

### LINKED ISSUE #7289: database is empty in INFORMATION_SCHEMA.task_runs after submit ctas task
### Steps to reproduce the behavior (Required)
```
CREATE TABLE `t1` (
                      `k1` INT,
                      `k2` INT
                    ) ENGINE=OLAP
                    DUPLICATE KEY(`k1`)
                    COMMENT "OLAP"
                    DISTRIBUTED BY HASH(`k1`) BUCKETS 3
                    PROPERTIES (
                    "replication_num" = "1"
                    );
SUBMIT TASK task1 AS CREATE TABLE t2 AS SELECT * FROM t1;
SELECT * FROM INFORMATION_SCHEMA.task_runs where task_name='task1';
```

### Expected behavior (Required)
database is not empty
### Real behavior (Required)
```
+--------------------------------------+-----------+---------------------+---------------------+---------+----------+---------------------------------------------------+------------+----------------------------------+
| QUERY_ID                             | TASK_NAME | CREATE_TIME         | FINISH_TIME         | STATE   | DATABASE | DEFINITION                                        | ERROR_CODE | ERROR_MESSAGE                    |
+--------------------------------------+-----------+---------------------+---------------------+---------+----------+---------------------------------------------------+------------+----------------------------------+
| 1bf08e6f-ec88-11ec-bbc9-00163e10c9d6 | task7     | 2022-06-15 16:49:47 | 2022-06-15 16:50:57 | SUCCESS |          | CREATE TABLE t2 AS SELECT * FROM t1               |          0 | NULL                             |
+--------------------------------------+-----------+---------------------+---------------------+---------+----------+---------------------------------------------------+------------+----------------------------------+
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.3-RELEASE 93b7fe8 |
+----------------------------+
```

### LINKED ISSUE #7503: query INFORMATION_SCHEMA.task_runs failed after drop database
### Steps to reproduce the behavior (Required)
```
CREATE DATABASE db1;
CREATE DATABASE db2;
USE DATABASE db1;
SUBMIT TASK task1 AS CREATE TABLE t1 AS SELECT sleep(1);
USE DATABASE db2;
DROP DATABASE db1;
SELECT * FROM INFORMATION_SCHEMA.task_runs;
```

### Expected behavior (Required)
`query ok`
### Real behavior (Required)
`ERROR 1064 (HY000): failed to call frontend service, host: unknown`
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.3-RELEASE b202d93 |
+----------------------------+
```

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7289
Fix #7503

## Problem Summary(Required) ：

1.ConnectContext should use task set database
2.database should not use getDb because db may drop
3.do not should default cluster to user, only show db name

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 11 (09.09%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|1|1|100.00%|

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 11 (09.09%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 1]
```diff
@@ -358,7 +357,11 @@ public TGetTaskInfoResult getTasks(TGetTasksParams params) throws TException {
             info.setCreate_time(task.getCreateTime() / 1000);
             // Now there are only MANUAL types of Tasks
             info.setSchedule("MANUAL");
-            info.setDatabase(task.getDbName());
+            String dbName = task.getDbName();
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:360]: ClusterNamespace.getNameFromFullName(databaseName)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|1|1|100.00%|

**Review by gengjun-git — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**